### ISSUE TITLE
G3-G4 wrap-up

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#8358]
- [ ] Add a ChangeLog entry for G3 and G4.
- [ ] Update user documentation: `docs/driver-only-builds.md`.
- [ ] Update any strategy document (`docs/architecture/psa-migration/*.md`) that needs updating.
- [ ] ~Fix `all.sh` component messages - https://github.com/Mbed-TLS/mbedtls/pull/8527#discussion_r1395374366~. This was already done as the last commit in the original PR: https://github.com/Mbed-TLS/mbedtls/pull/8527/commits/7afd9a466310bac6b3effa19fcab5535f3acd614
- (List to be completed if and when we notice any other small enough item - larger items should be their own issue.)

**Depends on:** every other G3 or G4 task in this EPIC :)

**Comment by valeriosetti** [comment]
Question about:
> Update any strategy document (docs/architecture/psa-migration/*.md) that needs updating

The upcoming G5 goal #8598 affects the only real dispatching component we have for AEADs which is `BLOCK_CIPHER`. Should I extend `docs/architecture/psa-migration/md-cipher-dispatch.md` here or instead do it in #6337?

**Comment by mpg** [comment]
Since this is the G3-G4 wrap-up issue, things about G5 are out of scope here, and will be covered by 6337 indeed.
